cpp/src/arrow/dataset/CMakeLists.txt

```diff
@@ -36,6 +36,10 @@ if(ARROW_PARQUET)
   string(APPEND ARROW_DATASET_PKG_CONFIG_REQUIRES " parquet")
 endif()
 
+if(PARQUET_REQUIRE_ENCRYPTION)
+  add_definitions(-DPARQUET_REQUIRE_ENCRYPTION=1)
+endif()
+
```

Comment:
Could you use `cpp/src/arrow/util/config.h.cmake` instead of `add_definitions()`?